### DISCUSSION TITLE
Simulation of molecule deposition in LAMMPS using MACE generated potential.

### DISCUSSION THREAD

**Discussion by nikhilwilsonk** [ACEsuit/mace#455] [Q&A] [answered]
Hi everyone,

I am currently trying to simulate a surface deposition of oxygen molecule using LAMMPS and the MACE-potential. I generated AIMD-NVT simulation data for three different system sizes of the system that I am trying to simulate (oxygen molecule deposited from a height onto the surface). After sampling few configurations of this generated data, I have generated a MACE model then a lammps potential using the mace_lammps_create_model.py.   

Afterwards, I have generated a larger supercell of the same facet system and performed lammps simulation.  The AIMD-NVT simulation results in the oxygen molecule not breaking on the surface but the LAMMPS simulation results in the molecule breaking and one oxygen atom rapidly moving across the surface.
Any idea as to why this happens or Any suggestions for troubleshooting this?

**Comment by ilyes319** [comment]
Hi @nikhilss219,

I would need to have a more detailed look at your training run to see if there is any problem. How did you select your configurations from the training set? In these situations, if you have a stable model, we would do some NEBs with the model and then evaluate it with DFT and add them to the training set. See Lars' paper for example (https://www.nature.com/articles/s41524-023-01124-2).

**Reply by nikhilwilsonk** [reply]
I selected my configurations by sorting the energies and then binning them equally, then random sampling the configurations from each bin. I am also curious on how to implement the iterative training and have been reading upon the literature. I will update here once I am able to implement that. I am also attaching my training log file.

Thanks for your time,
Nikhil
[training_log.txt](https://github.com/user-attachments/files/15895654/training_log.txt)

**Comment by gabor1** [comment, marked as answer]
If you only used AIMD configs, then I would expect that your trained potential finds configurations that are high energy in DFT (and so are not in your training set) but low energy in your fitted potential. So you need a round if "iterative training". Get a few configurations from your MACE MD, reevaluate them with DFT, add it to your training data, and continue the training from the previous potential. It's likely that you only need a few dozen configs and only one round of this iterative training. Pick configurations from near the time when interesting things happen (O2 splits, O2 traverses the surface)

**Reply by nikhilwilsonk** [reply]
Actually, the way I sampled configurations from the AIMD is by sorting the energies and binning them in equal groups, and then random sampling them. This was done such that there are configurations from all regions. Yes I am learning how to implement an iterative training process.

Thanks for taking your time.
Nikhil

PS: I am reading upon literature send by @ilyes319 from Prof Lars'  (https://www.nature.com/articles/s41524-023-01124-2)
